.github/actions/repo-gardening/src/tasks/check-description/index.js

```diff
@@ -0,0 +1,266 @@
+/**
+ * External dependencies
+ */
+const moment = require( 'moment' );
+
+/**
+ * Internal dependencies
+ */
+const debug = require( '../../debug' );
+const getLabels = require( '../../get-labels' );
+const getNextValidMilestone = require( '../../get-next-valid-milestone' );
+
+/* global GitHub, WebhookPayloadPullRequest */
+
+/**
+ * Check if a PR has unverified commits.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<boolean>} Promise resolving to boolean.
+ */
+async function hasUnverifiedCommit( octokit, owner, repo, number ) {
+	let isUnverified = false;
+
+	for await ( const response of octokit.paginate.iterator( octokit.pulls.listCommits, {
+		owner: owner.login,
+		repo,
+		pull_number: +number,
+	} ) ) {
+		response.data.map( commit => {
+			if ( commit.commit.message.includes( '[not verified]' ) ) {
+				isUnverified = true;
+			}
+		} );
+	}
+
+	return isUnverified;
+}
+
+/**
+ * Check for status labels on a PR.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<boolean>} Promise resolving to boolean.
+ */
+async function hasStatusLabels( octokit, owner, repo, number ) {
+	const labels = await getLabels( octokit, owner, repo, number );
+	let hasStatus = false;
+
+	// We're really only interested in status labels
+	labels.map( label => {
+		if ( label.includes( '[Status]' ) ) {
+			debug( `check-description: this label (${ label }) includes the word Status` );
+			hasStatus = true;
+		}
+	} );
+
+	return hasStatus;
+}
+
+/**
+ * Build a string with info about the next milestone.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<string>} Promise resolving to info about the next release for that plugin.
+ */
+async function buildMilestoneInfo( octokit, owner, repo, number ) {
+	const labels = await getLabels( octokit, owner, repo, number );
+
+	// Find out what plugin we need to worry about.
+	// We default to the Jetpack plugin for now.
+	let plugin;
+	labels.map( label => {
+		if ( label.includes( '[Plugin] Jetpack' ) ) {
+			plugin = 'jetpack';
+		}
+
+		if ( label.includes( '[Plugin] Beta Plugin' ) ) {
+			plugin = 'beta';
+		}
+
+		plugin = 'jetpack';
+	} );
```

Comment:
I see that this block is duplicated few times. It might be a good idea to extract it into its own method. I imagine we would tweak the list of plugins often enough.

Reply:
Good point. Done in 6e5aa858e2212acfaed785f7273535d5213a1dc1.